components/org.wso2.carbon.identity.application.authenticator.sessionauth/src/main/java/org/wso2/carbon/identity/application/authenticator/sessionauth/util/AuthenticatorUtil.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.wso2.carbon.identity.application.authenticator.sessionauth.util;
+
+import org.apache.axis2.transport.http.HTTPConstants;
+import org.apache.commons.httpclient.HttpStatus;
+import org.apache.commons.ssl.Base64;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2.carbon.identity.application.authentication.framework.config.ConfigurationFacade;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.application.authenticator.sessionauth.SessionCountAuthenticatorConstants;
+import org.wso2.carbon.identity.application.authenticator.sessionauth.exception.SessionValidationException;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Contains Utility methods used in the Session Count Authenticator.
+ */
+public class AuthenticatorUtil {
+
+    /**
+     * Method to retrieve session data from session data source.
+     *
+     * @param authenticatedUser AuthenticatedUser object that represent the user
+     * @return JSON array with each element describing active session
+     * @throws IOException                When it fails to read response from the REST call
+     * @throws SessionValidationException when REST response is not in state 200
+     */
+    public static JSONArray getSessionDetails(AuthenticatedUser authenticatedUser) throws
+            IOException, SessionValidationException {
+
+        HttpClientBuilder httpClientBuilder = HttpClientBuilder.create();
```

Comment:
We better use OSGI (java call) to get the respective session details. Calling HTTP service would not be ideal.

Reply:
As discussed offline, this is kept as it is and a siddhi app will be written to retrieve this data and called from here once the Identity Server Analytics is replaced with Stream Processor